### ISSUE TITLE
Fails to load more than one plugin

### ISSUE THREAD

**Issue by eraserhd** [neovim/node-host#21]
I'm trying to run the most recent version of `node-host` with some fairly recent `neovim`, and I have `pathogen.vim` and `clj-refactor.nvim` and `nvim-parinfer.js` plugins in my `bundle` directory.  Tried on `node` 6.7.0 and 7.0.0.

If I move either plugin out of the way, `:UpdateRemotePlugins` works.  Both together fail with the following:

```
function remote#host#UpdateRemotePlugins[6]..<SNR>14_RegistrationCommands, line 20
Vim(let):Channel was closed by the client
remote/host: generated rplugin manifest: /Users/jfelice/.local/share/nvim/rplugin.vim
Press ENTER or type command to continue
```

I can get both working together manually by splicing together the lines in `~/.local/share/nvim/rplugin.vim` produced by running it with each plugin installed by itself.

**Comment by justinmk** [comment]
seems like a bug in UpdateRemotePlugins

**Comment by mhartington** [comment]
Seeing this as well. Interestingly only with js plugins and not multiple python plugins

**Comment by nhynes** [comment]
Thanks @eraserhd and @mhartington for isolating the issue! I'll be able to take a look at this over the weekend. If it's blocking your work, however, I'll prioritize a fix.

**Comment by mhartington** [comment]
Not to be annoying and "that person", but did you get a chance to look at things @nhynes?

**Comment by nhynes** [comment]
Hey @mhartington, I think that it is *I* who is being "that person" by not having taken a serious look at this yet. I'm quite busy until the 16th,  but if I have a few free moments between now and then, I'll fix this instead of something else!

If, after the 16th, this isn't yet fixed, you're encouraged to call me a bad maintainer :)

Sorry for not being able to fix this faster!

**Comment by mhartington** [comment]
hehe, no worries! 

Happy to dive in and take a look, just not sure where to poke around first.

**Comment by nhynes** [comment]
I was without internet for a while yesterday, so I went and took a look at his bug. The issue is actually in [node-client](https://github.com/neovim/node-client/blob/master/index.js#L180). I think that he problem is that the whole API generation bit gets run in between the two `specs` requests and when the client detaches (to reattach again with a msgpack session that has the correct extypes) the second one never gets handled.

I fixed the multi-plugin bug by modifying [msgpack5rpc](https://github.com/tarruda/node-msgpack5rpc) to allow adding types after initialization and removing the whole detach-reattach block. Unfortunately, that introduced a new issue seems to prevent the host from attaching to nvim for non-`specs` requests. Debugging is complicated by the lack of a print channel, but I'll keep at it in my down time (and maybe clear a path for print statements, too)! 

Thanks for finding this bug! I can't believe it had gone unnoticed for so long :)
